salvo-remote/sandboxes/sandboxes.go

```diff
@@ -0,0 +1,198 @@
+// Package sandboxes manages the life cycle of a Salvo remote sandbox.
+package sandboxes
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/go-version"
+	"github.com/hashicorp/hc-install/product"
+	"github.com/hashicorp/hc-install/releases"
+	"github.com/hashicorp/terraform-exec/tfexec"
+)
+
+// sandboxDefDir is the directory where the sandbox Terraform definition files are located.
+const sandboxDefDir = "terraform/"
+
+// Type enumerates the types of sandboxes supported.
+type Type int
+
+// String implements fmt.Stringer()
+func (t Type) String() string {
+	if n, ok := typeNames[t]; ok {
+		return n
+	}
+	return "TypeUnknown"
+}
+
+// typeNames maps Type values to human readable names.
+var typeNames = map[Type]string{
+	TypeDefaultSandboxX64: "default_sandbox_x64_build_ids",
+}
+
+const (
+	typeUnknown Type = iota
```

Comment:
Just curious, is `typeUnknown` being used anywhere?

Reply:
`typeUnknown` would get used if someone passed in the default value of the type `sandboxes.Type`.

Defining it here helps in isolating that sort of programming mistake, same as when we have an enum in a proto file and we assign the first value to UNKNOWN.